### ISSUE TITLE
x/tools/gopls: stuck with error after resolving git conflict markers

### ISSUE THREAD

**Issue by atombender** [golang/go#38879]
#### What did you do?
Did a `git rebase` and resolved conflict markers in a file.

#### What did you expect to see?
No errors.

#### What did you see instead?
gopls thinks there are still conflict markers.

![Screen Shot 2020-05-05 at 12 30 21](https://user-images.githubusercontent.com/50314/81090744-30b71980-8ecc-11ea-9902-cadda014e0ee.jpg)

#### Log
[gopls.log.gz](https://github.com/golang/go/files/4582187/gopls.log.gz), scrubbed of `didOpen` sources, except for the file in question, `source_test.go`.

#### Build info

```
golang.org/x/tools/gopls master
    golang.org/x/tools/gopls@v0.0.0-20200501005904-d351ea090f9b h1:CAqIXUMd9mH3t2O1DRj2IWu0Zh75A4fWPA+YjJOmXdE=
    github.com/BurntSushi/toml@v0.3.1 h1:WXkYYl6Yr3qBf1K79EBnL4mak0OimBfB0XUf9Vl28OQ=
    github.com/sergi/go-diff@v1.1.0 h1:we8PVUC3FE2uYfodKH/nBHMSetSfHDR6scGdBi+erh0=
    golang.org/x/mod@v0.2.0 h1:KU7oHjnv3XNWfa5COkzUifxZmxp1TyI7ImMXqFxLwvQ=
    golang.org/x/sync@v0.0.0-20190911185100-cd5d95a43a6e h1:vcxGaoTs7kV8m5Np9uUNQin4BrLOthgV7252N8V+FwY=
    golang.org/x/tools@v0.0.0-20200430221153-f26c0dd9827d h1:8emZwJTManptbFnp36i5fbhO/1vUs3rfnh5e3Th8HEQ=
    golang.org/x/xerrors@v0.0.0-20191204190536-9bdfabe68543 h1:E7g+9GITq07hpfrRu66IVDexMakfv52eLZ2CXBWiKr4=
    honnef.co/go/tools@v0.0.1-2020.1.3 h1:sXmLre5bzIR6ypkjXCDI3jHPssRhc8KD/Ome589sc3U=
    mvdan.cc/xurls/v2@v2.1.0 h1:KaMb5GLhlcSX+e+qhbRJODnUUBvlw01jt4yrjFIHAuA=
```

#### Go info

```
go version go1.14 darwin/amd64

```

**Comment by stamblerre** [comment]
Seeing error messages like

```
[Error - 12:27:01 PM] 2020/05/05 12:27:01
	message="failed to load workspace packages, skipping diagnostics"
	error=github.com/sanity-io/gradient/pkg/search/elasticsearch/indexstate_test [github.com/sanity-io/gradient/pkg/search/elasticsearch/indexstate.test] has no metadata
	snapshot=244
	directory=file:///Users/alex/Projects/Sanity/gradient
```

which leads me to believe that this is a duplicate of https://github.com/golang/go/issues/38878, which is in turn a duplicate of https://github.com/golang/go/issues/38423 (and probably related to https://github.com/golang/go/issues/38358).

**Comment by atombender** [comment]
Got it. It's pretty hard to understand what the cause of a bug is and therefore know if it's a dupe, sorry. As an aside, is there no way for gopls to report itself as broken so that things don't get stuck, and maybe an error message bubbles up to VSCode?

**Comment by stamblerre** [comment]
No worries - I appreciate you filing all of these issues with logs. We can probably use a `window/showMessage` when we start skipping diagnostics to make the error more visible to the user.

**Comment by atombender** [comment]
I just got the error again, _after_ restarting gopls. In other words:

* `git rebase`, got conflict markers.
* Restarted gopls and even reloaded window a couple of times.
* Gopls reports same errors as top of this issue.

[gopls.log.gz](https://github.com/golang/go/files/4588819/gopls.log.gz)

**Comment by stamblerre** [comment]
Thanks for the report. It looks like we're failing to load a package from a file, but we don't log any errors. I added a better error log in https://golang.org/cl/230315, so when that's merged it would be easier to figure out why this happens.

**Comment by heschik** [comment]
I suspect this is a dupe of the transitive invalidation bug?

**Comment by stamblerre** [comment]
Yes, probably true. Should be fixed in `gopls/v0.4.4`.
